lib/dpdk/iceBoardVDIF.hpp

```diff
@@ -291,7 +291,13 @@ void iceBoardVDIF::copy_packet_vdif(struct rte_mbuf* mbuf) {
     int from_idx = header_offset + offset;
     int mbuf_len = mbuf->data_len;
     for (uint32_t time_step = 0; time_step < samples_per_packet; ++time_step) {
+        int output_idx_base = (vdif_frame_location * frame_size +           // Frame location in output buffer.
+			       vdif_header_len +        // Offset for the vdif header.
+			       tel.to_freq_id(encoded_id, 0) + // Offset for ARO as in tel.to_freq_id for _num_freq_per_stream = 128
```

Comment:
@andrerenard - this call to a method seemed to be the dominant reason the compiler couldn't make this optimization itself. In principle, one could calculate something like
```
freq0_offset = tel.to_freq_id(encoded_id, 0)
freq_step = 1024 // tel.num_freq_per_stream
```
and then loop to `freq < tel.num_freq_per_stream`. I think this covers the contents of the switch statement inside `to_freq_id` -- but is this worth it?

Reply:
Yeah we refactored the handling of the `tel.to_freq_id()` function after this code was written, so that's likely where this performance regression showed up.  In this mode `encoded_id` doesn't change at all for a given instance of `iceBoardVDIF`, and in fact if it does, I believe other parts of the system will reject those packets.   So you could just cache the output of this function.  There is also no way for this code to run with anything other than the FPGA in `shuffle16` mode, so hard coding the frequency generation, isn't really a problem here.   That might well be worth doing, since this function is called 390625 times/second even after moving it up here.